zipkin-storage/zipkin2_cassandra/src/main/java/zipkin2/storage/cassandra/internal/call/AccumulateAllResults.java

```diff
@@ -53,18 +57,15 @@
 
     abstract BiConsumer<Row, T> accumulator();
 
+    /** Iterates through the rows in each page, flatmapping on more results until exhausted */
     @Override public Call<T> map(ResultSet rs) {
-      if (!rs.isFullyFetched()) rs.fetchMoreResults(); // TODO: dropped future
-      for (Row row : rs) {
-        accumulator().accept(row, pendingResults());
-        // TODO magic number
-        if (2000 == rs.getAvailableWithoutFetching() && !rs.isFullyFetched()) {
-          rs.fetchMoreResults(); // TODO: dropped future
-        }
-        if (0 == rs.getAvailableWithoutFetching()) break;
```

Comment:
So FTR: without the pre-emptive asynchronous fetching of the next page, the parsing will stop and block every 5k rows while it fetches the next page. Queries that process more than 5k rows are going to be significantly slower (the network fetch is the slowest part).